lib/libmbedtls/mbedtls/library/ccm.c

```diff
@@ -284,23 +310,52 @@ static int ccm_auth_crypt( mbedtls_ccm_context *ctx, int mode, size_t length,
          * Increment counter.
          * No need to check for overflow thanks to the length check above.
          */
-        for( i = 0; i < q; i++ )
-            if( ++ctr[15-i] != 0 )
+        for( i = 0; i < ctx->q; i++ )
+            if( ++ctx->ctr[15-i] != 0 )
                 break;
     }
 
+    return 0;
+}
+
+int mbedtls_ccm_finish( mbedtls_ccm_context *ctx,
+                unsigned char *tag,
+                size_t tag_len )
+{
+    int ret;
+    unsigned char i;
+    size_t olen;
+
     /*
      * Authentication: reset counter and crypt/mask internal tag
      */
-    for( i = 0; i < q; i++ )
-        ctr[15-i] = 0;
+    for( i = 0; i < ctx->q; i++ )
+        ctx->ctr[15-i] = 0;
 
-    CTR_CRYPT( y, y, 16 );
-    memcpy( tag, y, tag_len );
+    CTR_CRYPT( ctx->y, ctx->y, 16 );
+    memcpy( tag, ctx->y, tag_len );
 
     return( 0 );
 }
 
+static int ccm_auth_crypt( mbedtls_ccm_context *ctx, int mode, size_t length,
+                           const unsigned char *iv, size_t iv_len,
+                           const unsigned char *add, size_t add_len,
+                           const unsigned char *input, unsigned char *output,
+                           unsigned char *tag, size_t tag_len )
+{
+    int ret = 0;
+
+    ret = mbedtls_ccm_starts( ctx, mode, length, iv, iv_len, add, add_len, tag_len );
+    if( ret != 0 )
+        return ret;
+    ret = mbedtls_ccm_update( ctx, length, input, output );
+    if( ret != 0 )
+        return ret;
+    ret = mbedtls_ccm_finish( ctx, tag, tag_len );
+        return( ret );
```

Comment:
Fix indentation for the last `return( ret )`.
Maybe add parentheses around `ret` in the `return ret;` occurrences above.

Reply:
Yes, I will fix it.